specs/agents/metrics-otel.md

```diff
@@ -0,0 +1,181 @@
+
+To summarize and to give an example, given the following set of attributes
+
+| **Attribute Key** | **Attribute Value** | **Type**       |
+|-------------------|---------------------|----------------|
+|  `foo.bar`        | `baz`               | `string`       |
+|  `my_array`       | `[first, second]`   | `string array` |
+|  `testnum*`       | `42.42`             | `number`       |
+
+the following labels must be used for the resulting metricset:
+
+| **Label Key** | **Label Value** | **Type**       |
+|---------------|-----------------|----------------|
+|  `foo_bar`    | `baz`           | `string`       |
+|  `testnum_`   | `42.42`         | `number`       |
+
+The OpenTelemetry specification allows the definition of metrics with the same name as long as they reside within a different instrumentation scope ([spec link](https://opentelemetry.io/docs/reference/specification/metrics/api/#get-a-meter)). In order to avoid conflicts, agents MUST add a mandatory label to the metricset with the key `otel_instrumentation_scope_name` and the corresponding instrumentation scope name as label value.
+
+## Exporter Installation
+
+The onboarding should be as easy as possible for users. Ideally, the users should be able to browse their OpenTelemetry metrics by just following the standard agent installation process without requiring additional configuration or code changes.
+
+We focus on two ways of how OpenTelemetry may be used by users:
+
+1. The user application has a OpenTelemetry Metrics SDK Instance setup and configured. E.g. the user already has a prometheus exporter running and did some customizations on their metrics via [Views](https://opentelemetry.io/docs/reference/specification/metrics/sdk/#view).
+2. The user application does not setup a OpenTelemetry Metrics SDK Instance, but uses the OpenTelemetry Metrics API to define metrics. This is usually the case when the user application does not care / know about OpenTelemetry metrics, but a library used by the application implemented observability via OpenTelemetry.
+
+For **1.** we want to make sure that the existing user configuration is respected by our apm agents. Ideally, agents SHOULD just register an additional exporter to the existing OpenTelemetry Metrics SDK instance(s). If the agent and language capabilities allow it, the exporter registration SHOULD be done automatically without requiring code changes by the user. 
+
+For **2.** agents MAY automatically register an agent-provided SDK instance to bind the user provided OpenTelemetry API to, if this is possible in their language and does not cause too much overhead of any kind (e.g. implementation or agent package size).
+Agents MUST NOT override a user-provided global OpenTelemetry metrics SDK with their own SDK.
```

Comment:
[minor] Maybe worth mentioning that this is different from the OTel bridge where the agent overrides the values returned by `GlobalOpenTelemetry` with its own implementation. Here with metrics SDK the agent just adds an extra exporter at the last step of the metrics pipeline (potentially replacing another sending metrics to another backend).

Reply:
> Maybe worth mentioning that this is different from the OTel bridge where the agent overrides the values returned by GlobalOpenTelemetry with its own implementation.

Isn't `GlobalOpenTelemetry` something java-specific?  It is just the java approach of binding the opentelemetry API to an implementation (e.g. an SDK or our bridge).

> Here with metrics SDK the agent just adds an extra exporter at the last step of the metrics pipeline (potentially replacing another sending metrics to another backend)

I don't quiet understand what you mean here? When we add our exporter, we don't replace existing exporters on the user-provided metrics SDK instance. Instead, both exporters live happily side by side.